Add option enclosure required

andriesreitsma opened the pull request:
### Requirements

Mark the following tasks as done:

* [x] Checked the codebase to ensure that your feature doesn't already exist.
* [x] Checked the pull requests to ensure that another person hasn't already submitted the feature or fix.
* [x] Adjusted the Documentation. (Not needed)
* [ ] Added tests to ensure against regression.

### Description of the Change

Add the newly created option to set requiring an enclosure for the CSV export to false, to prevent wrapping the cells with default value, since it's not nullable anymore.

### Why Should This Be Added?

Since version PHPSpreadSheet 1.13 the default value for `enclosure` is set to `"` and not nullable anymore.

### Benefits

Export to .csv can be done without enclosures again, so importing in older systems which are not adjustable are functioning as expected.

### Possible Drawbacks

None
### Verification Process

We do large exports for oldskool dos/windows applications, after this fix we can export without encloses again.

### Applicable Issues

andriesreitsma committed:
Update WriterFactory with new setEnclosureRequired

Since version PHPSpreadSheet 1.13 the default value for `enclosure` is set to `"` and not nullable anymore.

andriesreitsma committed:
Update Concern with enclosure_required

andriesreitsma committed:
Add default option to config

andriesreitsma committed:
Typo

andriesreitsma committed:
Style issue?

patrickbrouwers commented:
@atreitsma thanks for the PR, can you also bump the minimum PhpSpreadsheet version to 1.13?

If I understand your description correctly, this should work backwards compatible? Or if not, can we make it backwards compatible by default?

andriesreitsma committed:
Bump version phpspreadsheet to 1.14

function `SetEnclosureRequired` was introduced first in this release.

andriesreitsma commented:
> @atreitsma thanks for the PR, can you also bump the minimum PhpSpreadsheet version to 1.13?
> 
> If I understand your description correctly, this should work backwards compatible? Or if not, can we make it backwards compatible by default?

It does not yet, because bumping it to versions higher than `1.12` breaks the "enclosure" since setting only `""` it defaults to `"` as an enclosure. Guess I should add a function to validate the input, and if it's empty set require enclosure to false. Will update it and make it more clean... doing it remotely so forgive me the edits directly on github!

andriesreitsma committed:
Simplify setting enclosure required

For backwards compatibility this is enough, so setting `enclosure` to "" triggers the requirement of having an enclosure.

andriesreitsma committed:
Cleaning up after editing writer function.

andriesreitsma committed:
Cleaning up after function change

patrickbrouwers commented:
Thanks! No rush :) Can you add a test that validates that it's backwards compatible?

andriesreitsma commented:
Due to GitHub outage I'm missing a (style) commit in this pull request ... will get this sorted out, and will look into the tests later on which is totally not my expertise, so forgive me if it takes a bit longer.

andriesreitsma commented:
@patrickbrouwers just figured out you changed the test to pass only 3 days ago ;-) so if you revert that commit you should get a failing outcome in `tests/Concerns/WithCustomCsvSettingsTest.php` see your commit 6871c2b98176e3019a4a25376dbcfc0a088d1569 so if you add the code:

```php
    /**
     * @test
     */
    public function can_store_csv_export_without_enclosure()
    {
        $export = new class implements FromCollection, WithCustomCsvSettings {
            /**
             * @return Collection
             */
            public function collection()
            {
                return collect([
                    ['A1', 'B1'],
                    ['A2', 'B2'],
                ]);
            }

            /**
             * @return array
             */
            public function getCsvSettings(): array
            {
                return [
                    'delimiter'              => ';',
                    'enclosure'              => '',
                    'line_ending'            => PHP_EOL,
                    'use_bom'                => true,
                    'include_separator_line' => true,
                    'excel_compatibility'    => false,
                ];
            }
        };

        $this->SUT->store($export, 'custom-csv.csv');

        $contents = file_get_contents(__DIR__ . '/../Data/Disks/Local/custom-csv.csv');

        $this->assertStringContains('sep=;', $contents);
        $this->assertStringContains('A1;B1', $contents);
        $this->assertStringContains('A2;B2', $contents);
    }
```
Could add this to the request later on if you want!

patrickbrouwers commented:
Aha! That actually does explain why that test started failing. If you could revert my changes to that test, that would be great!

patrickbrouwers commented:
Thanks for the PR! i'll revert the test myself
